client/modules/ftml-components/src/components/code.ts

```diff
@@ -0,0 +1,88 @@
+import { highlight } from "@wikijump/prism"
+import { defineElement } from "../util"
+
+/**
+ * FTML `[[code]]` element. Automatically highlights the contents of its
+ * `<code>` child with Prism.
+ */
+export class Code extends HTMLPreElement {
+  static tag = "wj-code"
+
+  /** The language highlighting is being done with. */
+  declare language: string | null
+
+  /** The current textual contents of this element. */
+  declare content: string
+
+  /** The compiled/highlighted HTML. */
+  declare html?: string
+
+  constructor() {
+    super()
+
+    this.language = null
+    this.content = ""
+
+    // watch for changes to textual content
+    const mutationObserver = new MutationObserver(() => this.update())
+    mutationObserver.observe(this, { characterData: true, subtree: true })
+  }
+
+  /**
+   * Extracts the language to highlight with from this elements classes.
+   * Specifically, the `wj-language-{name}` class.
+   */
+  private getLanguageFromClass() {
+    const classes = Array.from(this.classList)
+    for (const name of classes) {
+      if (name.startsWith("wj-language-")) return name.substr(12)
```

Comment:
Maybe worth leaving a comment that this is always ASCII lowercase to avoid unnecessary changes in the future by people who don't remember.